### PR TITLE
Add TablestoreVectorStore.

### PR DESCRIPTION
**Pull request by xjtushilei**
Thank you for contributing to LangChain!

- [x] **PR title**:  community: add TablestoreVectorStore

- [x] **PR message**: 
    - **Description:** add TablestoreVectorStore
    - **Dependencies:** none

- [x] **Add tests and docs**: If you're adding a new integration, please include
  1. a test for the integration: yes
  2. an example notebook showing its use: yes

If no one reviews your PR within a few days, please @-mention one of baskaryan, efriis, eyurtsev, ccurme, vbarda, hwchase17.

### CODE REVIEW

File: docs/docs/integrations/providers/tablestore.mdx
Comment (baskaryan) [inline docs/docs/integrations/providers/tablestore.mdx:1]: if this is provided by alibaba cloud should this info be consolidated in docs/docs/integrations/alibaba_cloud.mdx?

Comment (xjtushilei) [reply docs/docs/integrations/providers/tablestore.mdx:1]: Thank you for your reminder. I have deleted the separate file and added it to `alibaba_cloud.mdx`.

**Comment by xjtushilei** [comment]
@baskaryan  Hello, I have already revised the previous comments. When you are free, can you help me take another look at the Review questions?

**Comment by xjtushilei** [comment]
@baskaryan   Hello, I have already revised the previous comments. When you are free, can you help me take another look at the Review questions?

**Comment by xjtushilei** [comment]
Hi， Any progress?

**Comment by xjtushilei** [comment]
This PR has not made any progress for a long time, so I updated it to the latest branch and modified some formatting issues according to CI. 

@baskaryan Please help merge this PR.

**Review by baskaryan — APPROVED** [review]

**Pull request merged**